src/faldbt/project.py

```diff
@@ -683,33 +712,51 @@ def _firestore_dict_to_document(data: Dict, key_column: str):
 
 
 def _map_nodes(
-    run_results: DbtRunResult, manifest: DbtManifest, generated_models: Dict[str, Path]
+    run_results: DbtRunResult,
```

Comment:
Is it too much to ask to refactor this function? Even splitting it into two would be helpful.

Reply:
It was merged before this PR. I separated them because now we have
- better type reporting in editor
- test check changes in the next PR for sources (https://github.com/fal-ai/fal/pull/405/files#diff-ba8ca12dab9905543efcdff4c5776ab3af382f7e6bd015cf68fb31032b8909bbR768)
- I think it may be better to refactor like Mederis suggesting, since we create nodes and then assign statuses